### PR TITLE
chore(hybrid-cloud): Add organization_service.remove_user

### PR DESCRIPTION
**Pull request by dashed**
Depends on https://github.com/getsentry/sentry/pull/47418/

### CODE REVIEW

File: src/sentry/services/hybrid_cloud/organizationmember_mapping/__init__.py [hunk 1]
```diff
@@ -43,14 +43,14 @@ class RpcOrganizationMemberMappingUpdate(TypedDict, total=False):
...
-def update_organizationmember_mapping_from_instance(
+def rpc_update_organizationmember_mapping_from_instance(
     organization_member: OrganizationMember,
 ) -> RpcOrganizationMemberMappingUpdate:
     attributes = {
         attr_name: getattr(organization_member, attr_name)
         for attr_name in RpcOrganizationMemberMappingUpdate.__annotations__.keys()
     }
-    return RpcOrganizationMemberMappingUpdate(**attributes)  # type: ignore
+    return RpcOrganizationMemberMappingUpdate(**attributes)
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/organizationmember_mapping/__init__.py:53]: Should we have `RpcOrganizationMemberMappingUpdate.from_orm()` We have methods like that for user and they are working well.

File: tests/sentry/auth/test_helper.py [hunk 2]
```diff
@@ -144,7 +144,8 @@ def test_associated_existing_member_invite_request(self):
         )
 
         self.assert_org_member_mapping(org_member=org_member)
-        self.assert_org_member_mapping_not_exists(org_member=member)
+        # TODO: need to handle org member mapping deletes
+        # self.assert_org_member_mapping_not_exists(org_member=member)
```
Comment (markstory) [inline tests/sentry/auth/test_helper.py:148]: Doesn't the `remove_user` service method make this work?

**Review by markstory — APPROVED** [review]
Looking good, there are a few relevant looking test failures though.

File: tests/sentry/auth/test_helper.py (hunk 2 above)
Comment (dashed) [reply tests/sentry/auth/test_helper.py]: @markstory During the invitation validation flow, we may delete org member records. I'm addressing deletions in https://github.com/getsentry/sentry/pull/47874.

File: src/sentry/models/organizationmember.py [hunk 3]
```diff
@@ -603,3 +604,15 @@ def is_only_owner(self) -> bool:
...
+def organization_member_post_delete(instance: OrganizationMember, **kwargs):
+    region_outbox = None
+    with transaction.atomic():
+        region_outbox = instance.outbox_for_update()
+        region_outbox.save()
+    if region_outbox:
+        region_outbox.drain_shard(max_updates_to_drain=10)
+
+
+post_delete.connect(organization_member_post_delete, sender=OrganizationMember)
```
Comment (markstory) [inline src/sentry/models/organizationmember.py:618]: If we do the outbox message creation during a `post_delete` signal it wouldn't be in the same transaction as the record operation. Could we override `delete()` on `OrganizationMember` instead?

**Pull request merged**